contracts/ExitHandler.sol

```diff
@@ -29,42 +29,392 @@ contract ExitHandler is DepositHandler {
+    address owner;
+    bool resolved;
+    bytes32[] proof;
   }
 
   uint256 public exitDuration;
+  uint256 public limboPeriod;
+  uint256 public piggybackStake;
+  uint256 public challengeStake;
   uint256 public exitStake;
   uint256 public nftExitCounter;
 
+  uint256 public constant LimboJoinDelay = (12 seconds);
+
   /**
    * UTXO → Exit mapping. Contains exits for both NFT and ERC20 colors
    */
   mapping(bytes32 => Exit) public exits;
+  mapping(bytes32 => LimboExit) public limboExits;
+  mapping(bytes22 => bool) public succesfulLimboExits;
 
   function initializeWithExit(
     Bridge _bridge, 
     uint256 _exitDuration, 
-    uint256 _exitStake) public initializer {
+    uint256 _exitStake,
+    uint256 _piggybackStake,
+    uint256 _challengeStake,
+    uint256 _limboPeriod) public initializer {
     initialize(_bridge);
     exitDuration = _exitDuration;
     exitStake = _exitStake;
+    challengeStake = _challengeStake;
+    piggybackStake = _piggybackStake;
+    limboPeriod = _limboPeriod;
     emit MinGasPrice(0);
   }
 
   function setExitStake(uint256 _exitStake) public ifAdmin {
     exitStake = _exitStake;
   }
 
+  function setPiggybackStake(uint256 _piggybackStake) public ifAdmin {
+    piggybackStake = _piggybackStake;
+  }
+
+  function setChallengeStake(uint256 _challengeStake) public ifAdmin {
+    challengeStake = _challengeStake;
+  }
+
   function setExitDuration(uint256 _exitDuration) public ifAdmin {
     exitDuration = _exitDuration;
   }
 
+  function startLimboExit(bytes memory inTxData, bytes32[] memory _youngestInputProof) public payable returns (bytes32 utxoId) { 
+
+    // exitor assumes tx to be canonical
+    require(msg.value >= exitStake, "Not enough ether sent to pay for exit stake");
+    TxLib.Tx memory transferTx = TxLib.parseTx(inTxData);
+    uint8 _outputIndex = 0;
+    
+    mapping(uint8 => LimboIn) storage inputs;
+    mapping(uint8 => LimboOut) storage outputs;
+
+    for (uint8 i =0; i < transferTx.outs.length; i++){
+      TxLib.Output memory outs = transferTx.outs[i];
+      outputs[i] = LimboOut({
+        amount: outs.value, 
+        owner: outs.owner, 
+        isPegged: false, 
+        color: outs.color, 
+        exitable: true,
+        challenge: {}
+        // utxoId: bytes32(uint256(i) << 120 | uint120(uint256(inTxHash))); });
+    });
+    for (uint8 i =0; i < transferTx.ins.length; i++){
+      TxLib.Input memory ins = transferTx.ins[i];
+      TxLib.Output memory out = transferTx.outs[ins.outpoint.pos];
+      inputs[i] = LimboIn({
+        amount: out.value,
+        owner: out.owner, 
+        isPegged: false, 
+        exitable: true,
+        challenge: {}
+      });
+    }
+    
+    bytes32 inTxHash = keccak256(inTxData);
+    bytes32 utxoId = bytes32(uint256(_outputIndex) << 120 | uint120(uint256(inTxHash)));
+    uint256 priority;
+    uint32 height;
+    uint32 timestamp;
+
+    // priority based on youngestInput
+    if (_youngestInputProof.length > 0) {
```

Comment:
unnecessary check? We always need youngestInputProof, no? Otherwise, I guess it will fail on line 171 (validateProof)